addons/mail/models/mail_message.py

```diff
@@ -659,9 +659,6 @@ def unlink(self):
         self.mapped('attachment_ids').filtered(
             lambda attach: attach.res_model == self._name and (attach.res_id in self.ids or attach.res_id == 0)
         ).unlink()
-        for elem in self:
```

Comment:
Would include a small explanation in the method docstring that we don't need to invalidate the related documents' cache as the unlink method invalidates the *whole* cache anyway (and also seems to flush everything?).

(I personally did not know).